relentless/optimize/method.py

```diff
@@ -179,17 +187,34 @@ def optimize(self, objective):
         while not converged and iter_num < self.max_iter:
             res = objective.compute()
             converged = self.has_converged(res)
+
             if not converged:
-                if isinstance(self.step_size, LineSearch):
-                    step = self.step_size.find(objective=objective, result_0=res)
-                else:
-                    step = self.step_size
+                if self.line_search is not None:
+                    for x in dvars:
+                        x.value = res.design_variables[x] - self.step_size*res.gradient(x)
+                    res_0 = objective.compute()
+                    res = self.line_search.find(objective=objective, res_start=res_0, res_end=res)
```

Comment:
So the default behavior for steepest descent + line search will have a search interval that changes magnitude throughout the optimization (the interval will be large for early iterations and small for late iterations). That is absolutely fine with me, but I think the more common use case of steepest descent + line search will be to have the search intervals all have the same magnitude. This'll let you take larger steps in the late iterations without being limited by numerical stability at early iterations (which I think will win out performance-wise).

I know we've talked about this a ton and I don't want to belabor the point, just want to make sure we all agree that this is the default behavior we want!